Hacking markdown and Makefile cleanup

sudo-bmitch opened the pull request:
The hacking.md doc had some stale references that needed to the cleaned. That resulted in the following changes:

- Remove/replace stale references from hacking doc.
- Add spacing for markdown.
- Add code block language for syntax highlighting.
- Remove leading "$ " from commands without output.
- Move links to end.
- Switch to automatically generated help in Makefile.

sajayantony reviewed (commented):
Do you want to remove the deprecated linters and just have `unused` 

```
WARN [runner] The linter 'deadcode' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused.
WARN [runner] The linter 'varcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused.
WARN [runner] The linter 'structcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused.
```

sudo-bmitch commented:
> Do you want to remove the deprecated linters and just have `unused` 
> 
> ```
> WARN [runner] The linter 'deadcode' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused.
> WARN [runner] The linter 'varcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused.
> WARN [runner] The linter 'structcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter. Replaced by unused.
> ```

That got fixed in #1088.

vbatts commented on Makefile:
```diff
@@ -41,25 +41,16 @@ DOC_FILES := \
-
-fmt:
+.PHONY: fmt
+fmt: ## format the json with indentation
```
meh. I don't _love_ having comments on the same line ...

sudo-bmitch replied:
I've got the same hesitation, and would really like it if they could go on the previous line. But the awk command everyone copies for `make help` depends on that syntax.
